core/vtk/ttkMetricDistortion/ttkMetricDistortion.cpp

```diff
@@ -0,0 +1,250 @@
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkTable");
+    info->Set(vtkAlgorithm::INPUT_IS_OPTIONAL(), 1);
+  } else
+    return 0;
+  return 1;
+}
+
+/**
+ * Specify the data object type of each output port
+ *
+ * This method specifies in the port information object the data type of the
+ * corresponding output objects. It is possible to either explicitly
+ * specify a type by adding a vtkDataObject::DATA_TYPE_NAME() key:
+ *
+ *      info->Set( vtkDataObject::DATA_TYPE_NAME(), "vtkUnstructuredGrid" );
+ *
+ * or to pass a type of an input port to an output port by adding the
+ * ttkAlgorithm::SAME_DATA_TYPE_AS_INPUT_PORT() key (see below).
+ *
+ * Note: prior to the execution of the RequestData method the pipeline will
+ * initialize empty output data objects based on this information.
+ */
+int ttkMetricDistortion::FillOutputPortInformation(int port,
+                                                   vtkInformation *info) {
+  if(port == 0) {
+    info->Set(ttkAlgorithm::SAME_DATA_TYPE_AS_INPUT_PORT(), 0);
+    return 1;
+  }
+  return 0;
+}
+
+/**
+ * Pass VTK data to the base code and convert base code output to VTK
+ *
+ * This method is called during the pipeline execution to update the
+ * already initialized output data objects based on the given input
+ * data objects and filter parameters.
+ *
+ * Note:
+ *     1) The passed input data objects are validated based on the information
+ *        provided by the FillInputPortInformation method.
+ *     2) The output objects are already initialized based on the information
+ *        provided by the FillOutputPortInformation method.
+ */
+int ttkMetricDistortion::RequestData(vtkInformation *ttkNotUsed(request),
+                                     vtkInformationVector **inputVector,
+                                     vtkInformationVector *outputVector) {
+
+  // --------------------------------------------------------------------------
+  // Get input object from input vector
+  // --------------------------------------------------------------------------
+  // Load Surface
+  vtkPolyData *inputSurface = vtkPolyData::GetData(inputVector[0]);
+  if(!inputSurface)
+    return 0;
+  auto noPoints = inputSurface->GetNumberOfPoints();
+  std::vector<std::vector<double>> surfacePoints(
+    noPoints, std::vector<double>(3));
+  for(unsigned int i = 0; i < noPoints; ++i) {
+    double point[3];
+    inputSurface->GetPoints()->GetPoint(i, point);
+    for(unsigned int j = 0; j < 3; ++j)
+      surfacePoints[i][j] = point[j];
+  }
+  auto noCells = inputSurface->GetNumberOfCells();
+  std::vector<std::vector<int>> surfaceCells(noCells);
+  for(unsigned int i = 0; i < noCells; ++i) {
+    auto noCellPoints = inputSurface->GetCell(i)->GetNumberOfPoints();
+    surfaceCells[i] = std::vector<int>(noCellPoints);
+    for(unsigned int j = 0; j < noCellPoints; ++j)
+      surfaceCells[i][j] = inputSurface->GetCell(i)->GetPointId(j);
+  }
+
+  // Detect boundary
```

Comment:
Same here, the triangulation class already implements many triangulation traversal features, including boundary access.
See `core/vtk/ttkScalarFieldCriticalPoints` and `core/base/scalarFieldCriticalPoints`.

Reply:
Done in the next commit.